File: api-chi/cmd/services/movie_depend_movie_category_test.go
package services

import (
	"api-chi/cmd/models"

	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_MovieDependMovieCategoryService(t *testing.T) {
	id := ""
	service := MovieDependMovieCategoryService{}

	t.Run("Count success", func(t *testing.T) {
		// Connect database
		err := service.Open()
		defer service.Close()
		assert.NoError(t, err)

		// Count database
		count, err := service.Count()
		assert.NoError(t, err)
		assert.Greater(t, count, 0)
	})

	t.Run("Create success", func(t *testing.T) {
		// Connect database
		err := service.Open()
		defer service.Close()
		assert.NoError(t, err)

		// Declare input
		input := models.MovieDependMovieCategory{
			MovieCategoryId: "360d60a6-3f3a-4c00-927a-04aa35410acc",
			MovieId:         "0bc66ff1-15ec-4d54-8830-1baf807b8091",
		}

		// Create database
		value, err := service.Create(&input)
		assert.NoError(t, err)
		assert.NotEmpty(t, value)

		// Assign value to id
		id = value
	})

	t.Run("Create same value must be failed prevent by trigger", func(t *testing.T) {
		// Connect database
		err := service.Open()
		defer service.Close()
		assert.NoError(t, err)

		// Declare input
		input := models.MovieDependMovieCategory{
			MovieCategoryId: "360d60a6-3f3a-4c00-927a-04aa35410acc",
			MovieId:         "0bc66ff1-15ec-4d54-8830-1baf807b8091",
		}

		// Create database
		_, err = service.Create(&input)
		assert.Error(t, err)
	})

	t.Run("Remove success", func(t *testing.T) {
		// Connect database
		err := service.Open()
		defer service.Close()
		assert.NoError(t, err)

		// Remove database
		value, err := service.Remove(&id)
		assert.NoError(t, err)
		assert.NotEmpty(t, value)
	})
}

File: api-fastify/src/tests/services/movie_depend_movie_category.test.ts
import MovieDependMovieCategoryService from "@/api/services/movie_depend_movie_category"
import { EnumMessage, ReturnMessage } from "@/types/message"
import { MovieDependMovieCategory } from "@/types/movie_depend_movie_category"

// Define test users
let currentId: string = ""

describe("MovieDependMovieCategoryService Integration Tests", () => {
  let service: MovieDependMovieCategoryService

  beforeAll(async () => {
    service = new MovieDependMovieCategoryService()
  })

  describe("Create", () => {
    it("should create a movie depend movie category successfully", async () => {
      const testCreated: MovieDependMovieCategory = {
        id: "",
        movieCategoryId: "360d60a6-3f3a-4c00-927a-04aa35410acc",
        movieId: "0bc66ff1-15ec-4d54-8830-1baf807b8091",
      }
      const result: ReturnMessage = await service.Create(testCreated)
      expect(result.success).toBe(true)
      expect(result.message).toBe(EnumMessage.CREATE_SUCCESS)
      expect(result.data).not.toBeUndefined()
      expect(typeof result.data.id).toBe("string")
      expect(typeof result.data.movieCategoryId).toBe("string")
      expect(typeof result.data.movieId).toBe("string")
      expect(result.data.movieCategoryId).toBe(testCreated.movieCategoryId)
      expect(result.data.movieId).toBe(testCreated.movieId)
      currentId = result.data.id
    })

    it("should create a movie depend movie category failed", async () => {
      const testCreated: MovieDependMovieCategory = {
        id: "",
        movieCategoryId: "360d60a6-3f3a-4c00-927a-04aa35410acc",
        movieId: "0bc66ff1-15ec-4d54-8830-1baf807b8091",
      }
      const result: ReturnMessage = await service.Create(testCreated)
      expect(result.success).toBe(false)
      expect(result.message).toBe(EnumMessage.CREATE_FAILED)
      expect(result.data).toBeUndefined()
    })
  })

  describe("Remove", () => {
    it("should remove a movie depend movie category successfully", async () => {
      const result: ReturnMessage = await service.Remove(currentId)
      expect(result.success).toBe(true)
      expect(result.message).toBe(EnumMessage.REMOVE_SUCCESS)
      expect(result.data).not.toBeUndefined()
    })
  })
})
